### PR TITLE
document basic build process

### PR DESCRIPTION
**Pull request by blackmad**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,36 +6,40 @@ XC_ARGS := -project $(PROJECT).xcodeproj -scheme $(SCHEME) -destination $(DESTIN
 XC_BUILD_ARGS := ONLY_ACTIVE_ARCH=NO OTHER_CFLAGS="-fembed-bitcode"
 XC_TEST_ARGS := GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES
 
-bootstrap:
+bootstrap: ## downlaod & install deps
```
Comment (jsani-radar) [inline Makefile]: nit: spelling

File: Makefile [hunk 2]
```diff
@@ -6,36 +6,40 @@ XC_ARGS := -project $(PROJECT).xcodeproj -scheme $(SCHEME) -destination $(DESTIN
...
-build:
+build: ## build module
 	xcodebuild $(XC_ARGS) $(XC_BUILD_ARGS)
 
-lint:
+lint: ## lint (all?) code in repo
 	pod lib lint --verbose
 
-format:
+format: ## format (all?) code in repo
```
Comment (jsani-radar) [inline Makefile]: @pingxia0823 can you confirm? I think this formats the code in `RadarSDK/`

File: Makefile [hunk 3]
```diff
@@ -6,36 +6,40 @@ XC_ARGS := -project $(PROJECT).xcodeproj -scheme $(SCHEME) -destination $(DESTIN
...
 	jazzy
 
 dist: clean-pretty format test-pretty build-pretty docs lint
 
-.PHONY: bootstrap clean test build lint format docs dist
+# taken from https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: bootstrap clean test build lint format docs dist list
```
Comment (jsani-radar) [inline Makefile]: is `.PHONY` meant to be at the bottom or top of a `Makefile`? I've seen both while here..

File: Makefile (hunk 2 above)
Comment (pingxia0823) [reply Makefile]: right this will lint all .h and .m files in repo

File: Makefile [hunk 4]
```diff
@@ -6,36 +6,40 @@ XC_ARGS := -project $(PROJECT).xcodeproj -scheme $(SCHEME) -destination $(DESTIN
...
+test: ## run all tests
 	xcodebuild test $(XC_ARGS)
 
-build:
+build: ## build module
 	xcodebuild $(XC_ARGS) $(XC_BUILD_ARGS)
 
-lint:
+lint: ## lint (all?) code in repo
 	pod lib lint --verbose
```
Comment (pingxia0823) [inline Makefile]: I think so

File: Makefile (hunk 1 above)
Comment (blackmad) [reply Makefile]: fixed

File: Makefile (hunk 2 above)
Comment (blackmad) [reply Makefile]: changed

File: Makefile (hunk 3 above)
Comment (blackmad) [reply Makefile]: this is where it was originally. I don't think it matters where it occurs

**Comment by blackmad** [comment]
given that this is in a public repo, can I get a +1 from @nickpatrick or @tjulien ?

**Comment by nickpatrick** [comment]
@blackmad Thanks for this! I wrote an SDK release process doc in Notion this morning that breaks down the `Makefile`: https://www.notion.so/radarlabs/SDK-release-process-c4cbb313360b45d9a4dd14803e43f8de

I think we want the main CTA in the public SDK repos to be to (1) read our docs and (2) install the pre-compiled SDK from CocoaPods or Carthage (https://radar.io/documentation/sdk-v3#ios-add). My concern with adding `Makefile` info to the `README` is that customers might get confused and think they need to compile the SDK themselves to use it, which we almost never want them doing.

I do like the idea of adding comments to the `Makefile`, though, and the `help` task is slick.

The `lint` task runs `pod lib lint`, which lints the `RadarSDK.podspec` file. We could rename it to `pod-lint` if we think it's clearer.

cc @tjulien

**Comment by blackmad** [comment]
other suggestion would be to put this in a like DEVELOP.md or something,
and link to that in a small way from the readme?

**Comment by nickpatrick** [comment]
Yep, I like that approach. I think Stripe does a good job of this, with:
- a `Releases` section (https://github.com/stripe/stripe-ios#releases) that instructs you to use CocoaPods or Carthage
- a `Getting Started` section (https://github.com/stripe/stripe-ios#getting-started) that points you at the docs
- a `Contributing` section (https://github.com/stripe/stripe-ios#contributing) with a CTA that is basically: "We welcome your contributions. Please open issues for feature requests or large changes. Please open PRs for small fixes. Before you open a PR, here's how you compile and run tests."

I'd want to put some thought into the copy. Happy to own that and push something with our next beta release.

**Comment by blackmad** [comment]
I'll clubhouse you + close this?

**Comment by nickpatrick** [comment]
Perfect, thanks

**Comment by nickpatrick** [comment]
Handling this in a separate PR

**Pull request closed without merging**